rclrs/src/context.rs

```diff
@@ -107,4 +107,46 @@ impl Context {
         // SAFETY: No preconditions for this function.
         unsafe { rcl_context_is_valid(handle as *mut _) }
     }
+
+    /// Polls the node for new messages and executes the corresponding callbacks.
+    ///
+    /// See [`WaitSet::wait`] for the meaning of the `timeout` parameter.
+    ///
+    /// This may under some circumstances return
+    /// [`SubscriptionTakeFailed`][1] when the wait set spuriously wakes up.
+    /// This can usually be ignored.
+    ///
+    /// [1]: crate::SubscriberErrorCode
+    pub fn spin_once(&self, node: &Node, timeout: Option<Duration>) -> Result<(), RclReturnCode> {
```

Comment:
I'll elaborate more later, but the `spin` functions can't be in the context. The context is only used for sharing data between nodes, but once the executors are implemented, spin will be moved there.

Reply:
Ok, I moved them into a new `executor` module (`pub use`d from lib.rs).

Reply:
Not sure about that, there's no executor infrastructure yet, so having an `executor` module is a bit misleading. I'd rather those functions are moved back to `lib.rs`, like other bindings do (e.g. [rclpy](https://github.com/ros2/rclpy/blob/master/rclpy/rclpy/__init__.py#L184=))

Reply:
Moved back to `lib.rs`.